src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Mapping/RazorMapToDocumentEditsEndpointTest.cs

```diff
@@ -0,0 +1,559 @@
+        {|mapUsing:using System;|}
+        {|mapUsing2:using System.Collections.Generic;|}
+        class MyComponent : ComponentBase
+        {
+            {|map1:public int Counter { get; set; }|}
+        }
+        """,
+        razorSource: """
+        @page "/counter"
+        {|mapUsing:@using System|}
+        {|mapUsing2:@using System.Collections.Generic|}
+
+        @code {
+            {|map1:public int Counter { get; set; }|} 
+        }
+        """,
+        newCSharpSource: """
+        using OtherNamespace;
+        using System;
+
+        class MyComponent : ComponentBase
+        {
+            public int NewCounter { get; set; }
+        }
+        """,
+        expectedRazorSource: """
+        @page "/counter"
+        @using System
+        @using OtherNamespace
+
+        @code {
+            public int NewCounter { get; set; } 
+        }
+        """);
+
+    [Fact]
+    public Task AddUsing_OrdersSystemCorrectly()
+    => TestAsync(
+        csharpSource: """
+        {|mapUsing:using System;|}
+        {|mapUsing2:using MyNamespace;|}
+        class MyComponent : ComponentBase
+        {
+            {|map1:public int Counter { get; set; }|}
+        }
+        """,
+        razorSource: """
+        @page "/counter"
+        {|mapUsing:@using System|}
+        {|mapUsing2:@using MyNamespace|}
+
+        @code {
+            {|map1:public int Counter { get; set; }|} 
+        }
+        """,
+        newCSharpSource: """
+        using OtherNamespace;
+        using System;
+        using System.Collections.Generic;
+
+        class MyComponent : ComponentBase
+        {
+            public int NewCounter { get; set; }
+        }
+        """,
+        expectedRazorSource: """
+        @page "/counter"
+        @using System
+        @using System.Collections.Generic
+        @using OtherNamespace
+
+        @code {
+            public int NewCounter { get; set; } 
+        }
+        """);
+
+    [Fact]
+    public Task UsingIndentation_DoesNotApply()
+        => TestAsync(
+        csharpSource: """
+        {|mapUsing:using System;|}
+        {|mapUsing2:using System.Collections.Generic;|}
+        class MyComponent : ComponentBase
+        {
+            {|map1:public int Counter { get; set; }|}
+        }
+        """,
+        razorSource: """
+        @page "/counter"
+        {|mapUsing:@using System|}
+        {|mapUsing2:@using System.Collections.Generic|}
+
+        @code {
+            {|map1:public int Counter { get; set; }|} 
+        }
+        """,
+        newCSharpSource: """
+            using System;
+            using System.Collections.Generic;
+
+        class MyComponent : ComponentBase
+        {
+            public int NewCounter { get; set; }
+        }
+        """,
+        expectedRazorSource: """
+        @page "/counter"
+        @using System
+        @using System.Collections.Generic
+
+        @code {
+            public int NewCounter { get; set; } 
+        }
+        """);
+
```

Comment:
Would be good to add tests for multiple groups of usings, since a bunch of the code is dedicated to that scenario. Also for `@using Goo = Bar` mixed in a bit.

Reply:
Added a few more, and a blocking item of removing `@using Goo = Bar` with a skipped test https://github.com/dotnet/razor/issues/11168